setup.py

```diff
@@ -826,7 +822,7 @@ def is_package_excluded(package: str, exclusion_list: List[str]):
 
 EXTRAS_REQUIREMENTS.update(
     {
-        'all': devel_all,
```

Comment:
I'm afraid this can be a breaking change for some installations

Reply:
It definitely can. It definitely needs UPDATING.md and MOVE_TO_2.0 updates.

We are using the "[all]" already in a few places where we expect also the "devel" to be installed (just look for "[all]") 

For example in:

* dev/provider_packages/README.md
* script/in_container/_in_container_utils.sh
* in scrpts in "scripts/ci/provider_packages/*.sh"  (but I am working on it now in #12512)

Reply:
Do we need there devel dependencies?